consumer/consumer_internal_test.go

```diff
@@ -373,126 +238,6 @@ func TestMessageConverterV1(t *testing.T) {
-	MarkOffsetCount int
-}
-
-// MarkOffset will increment the offset on the message and keep count
-// of how many times it has been called.
-func (m *mockOffsetStash) MarkOffset(msg *sarama.ConsumerMessage, metadata string) {
-	m.MarkOffsetCount++
-	msg.Offset++
-}
-
-// The mockHighWaterMarker implements the highWaterMarker interface
-// for testing purposes.
-type mockHighWaterMarker struct {
-	HighWaterMarkOffsetCount int
-}
-
-// HighWaterMarkOffset is required by the highWaterMarker interface.
-// We count the number of times this function is called, and return
-// that number as its result.
-func (m *mockHighWaterMarker) HighWaterMarkOffset() int64 {
-	m.HighWaterMarkOffsetCount++
-	return int64(m.HighWaterMarkOffsetCount)
-}
-
-// metricsHook implements the MetricsHook interface for testing purposes
-type metricsHook struct {
-	ReportCount int
-	t           *testing.T
-	testCase    func(msg Message, metadatas *Metrics) (string, func(*testing.T))
-}
-
-// Report counts how many times it has been called, and executes any
-// testCase that has been added to the metricsHook.  This allows us to
-// reuse the metricsHook type every time we want to make assertions
-// about the information it has been provided.
-func (mmh *metricsHook) Report(msg Message, metadatas *Metrics) {
-	mmh.ReportCount++
-	mmh.t.Run(mmh.testCase(msg, metadatas))
-}
-
-// testHandler implements the Handler interface for testing
-// purposes. It can be used to make assertions about the message
-// passed to HandleMessage.
-type testHandler struct {
-	t *testing.T
-	// testCase optionally holds a function that, given a message,
-	// returns a test name and a function to run the test.
-	testCase  func(*Message) (string, func(t *testing.T))
-	CallCount int
-}
-
-// HandleMessage will keep a count of how many times it is called and,
-// if a testCase is set on the testHandler, it will run it with the
-// message that HandleMessage received, allowing us to make assertions
-// about the nature of that message.
-func (h *testHandler) HandleMessage(m *Message) error {
-	h.CallCount++
-	if h.testCase != nil {
-		// We use testing.T.Run because we can easily see that
-		// the testCase has been run by looking at the verbose
-		// output of go test.
-		h.t.Run(h.testCase(m))
-	}
-	return nil
-}
-
-// Serve emits logs when it cannot create a new consumer
-func TestServeLogsErrorFromNewConsumer(t *testing.T) {
-	tl := NewTestLogger(t)
-	c := &Consumer{
-		config: newConfig(),
-		Logger: tl.Logger,
-	}
-	c.newConsumer = func(addrs []string, groupID string, topics []string, config *cluster.Config) (clusterConsumer, error) {
-		return nil, fmt.Errorf("oh noes! it doesn't work! ")
-	}
-	c.Handle("foo", MessageConverterV1(NewConfig("")), HandlerFunc(func(m *Message) error {
-		return nil
-	}))
-	err := c.Serve(NewConfig("some-id"), "foo")
-	require.Error(t, err)
-}
-
-// checks if the consumer validates the configuration correctly.
-func TestValidateConfig(t *testing.T) {
```

Comment:
Why has this test been removed?

Reply:
Because I removed `validateConfig` which was testing if the group mode was correctly set.

But this config exists only in the `sarama-cluster` dependency which is removed.